vm_insnhelper.c

```diff
@@ -287,7 +287,51 @@ vm_check_canary(const rb_execution_context_t *ec, VALUE *sp)
 #define vm_check_frame(a, b, c, d)
 #endif /* VM_CHECK_MODE > 0 */
 
-static inline rb_control_frame_t *
+#if USE_DEBUG_COUNTER
+static void
+vm_push_frame_debug_counter_inc(
+    const struct rb_executoon_context_struct *ec,
+    const struct rb_control_frame_struct *reg_cfp,
+    VALUE type)
+{
+    const struct rb_control_frame_struct *prev_cfp = RUBY_VM_PREVIOUS_CONTROL_FRAME(reg_cfp);
+
+    RB_DEBUG_COUNTER_INC(frame_push);
+
+    if (RUBY_VM_END_CONTROL_FRAME(ec) != prev_cfp) {
+        const bool curr = VM_FRAME_RUBYFRAME_P(reg_cfp);
+        const bool prev = VM_FRAME_RUBYFRAME_P(prev_cfp);
+        const enum rb_debug_counter_type pat[2][2] = {
+            { RB_DEBUG_COUNTER_frame_R2R, RB_DEBUG_COUNTER_frame_R2C, },
```

Comment:
These names should be hidden.
Previous code is easy to read for me.

Reply:
Hmm OK.  Let me revert that part.

Reply:
https://github.com/ruby/ruby/pull/3296/commits/55a35649bd2351510087d06c683da802466f5338